js/pihole/settings.js

```diff
@@ -0,0 +1,70 @@
+$(function () {
+  $("[data-mask]").inputmask();
+});
+
+$(".confirm-reboot").confirm({
+	text: "Are you sure you want to send a reboot command to your Pi-Hole?",
+	title: "Confirmation required",
+	confirm(button) {
+		$.post( "php/savesettings.php", { "field": "reboot" } );
+	},
+	cancel(button) {
+		// nothing to do
+	},
+	confirmButton: "Yes, reboot",
+	cancelButton: "No, go back",
+	post: true,
+	confirmButtonClass: "btn-danger",
+	cancelButtonClass: "btn-success",
+	dialogClass: "modal-dialog modal-mg" // Bootstrap classes for mid-size modal
+});
+
+$(".confirm-restartdns").confirm({
+	text: "Are you sure you want to send a restart command to your DNS server?",
+	title: "Confirmation required",
+	confirm(button) {
+		$.post( "php/savesettings.php", { "field": "restartdns" } );
+	},
+	cancel(button) {
+		// nothing to do
+	},
+	confirmButton: "Yes, restart DNS",
+	cancelButton: "No, go back",
+	post: true,
+	confirmButtonClass: "btn-danger",
+	cancelButtonClass: "btn-success",
+	dialogClass: "modal-dialog modal-mg"
+});
+
+$(".confirm-flushlogs").confirm({
+	text: "By default, the log is flushed at the end of the day via cron, but a very large log file can slow down the Web interface, so flushing it can be useful. Note that your statistics will be reset and you lose the statistics up to this point. Are you sure you want to flush your logs?",
+	title: "Confirmation required",
+	confirm(button) {
+		$.post( "php/savesettings.php", { "field": "flushlogs" } );
```

Comment:
On all of these POSTs from JS, wouldn't the PHP die at line 5 because it's not being called from settings.php?